### PR TITLE
New Post: Chromium Lands Flexbox Gap

### LINKED ISSUE #2323: content: flex gap
* Authors
  * @argyleink  
* Deadline: 2020 June 1 (Soft)
* [Draft](https://docs.google.com/document/d/12103CEAaDnCGNTjsD8QphBZYUgMu4A9lSGQp4q4GnWs/edit#heading=h.92s338ur0tz7)
* Related content
  * https://developer.mozilla.org/en-US/docs/Web/CSS/gap
  * https://codepen.io/argyleink/pen/abbVqEv
  * https://storage.googleapis.com/atoms-sandbox.google.com.a.appspot.com/css-at-cds/flex-n-grid-gap.mp4
* Stakeholders
  * @surma

### PR DESCRIPTION
**Pull request by argyleink**
Fixes #2323 

Deadline: May 27, 2020 (Hard)

- [x] Co-author to add their data: David Grogan
- [x] Hero image
- [x] `webm` conversion to the `.mp4`
- [ ] release alignment with Chromium (TBD)

### CODE REVIEW

File: src/site/content/en/blog/flexbox-gap/index.md [hunk 1]
```diff
@@ -0,0 +1,222 @@
+---
+title: Chromium Lands Flexbox `gap`
+subhead: Chromium's CSS Flexbox and Multi-Column layout engines now support gap. `grid-gap` is old news.
```
Comment (kaycebasques) [inline src/site/content/en/blog/flexbox-gap/index.md:3]:
```suggestion
subhead: Chromium's CSS Flexbox and Multi-Column layout engines now support `gap`. `grid-gap` is old news. 
```

File: src/site/content/en/blog/flexbox-gap/index.md [hunk 2]
```diff
@@ -0,0 +1,222 @@
+---
+title: Chromium Lands Flexbox `gap`
+subhead: Chromium's CSS Flexbox and Multi-Column layout engines now support gap. `grid-gap` is old news. 
+authors:
+  - adamargyle
+  # - dgrogan
+description: Chromium's CSS Flexbox and Multi-Column layout engines now support gap. `grid-gap` is old news.
```
Comment (kaycebasques) [inline src/site/content/en/blog/flexbox-gap/index.md:7]:
```suggestion
description: Chromium's CSS Flexbox and Multi-Column layout engines now support `gap`. `grid-gap` is old news. 
```

File: src/site/content/en/blog/flexbox-gap/index.md [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+  # - dgrogan
+description: Chromium's CSS Flexbox and Multi-Column layout engines now support gap. `grid-gap` is old news. 
+date: 2020-04-30
+updated: 2020-04-30
+tags:
+  - layout
+  - flexbox
+  - css
+  - gap
+# hero: hero.jpg
```
Comment (kaycebasques) [inline src/site/content/en/blog/flexbox-gap/index.md:15]: We use unsplash.com for most of our hero images

File: src/site/content/en/blog/flexbox-gap/index.md [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+At the time of writing `gap` is supported in desktop Firefox (version 63), Firefox for Android (version 63), and Chromium-based browsers (version 85). See [Browser compatibility](https://developer.mozilla.org/docs/Web/CSS/gap#Browser_compatibility) for updates.
+
+```css/3-4/1-2
+.grid {
+  display: grid;
+  grid-gap: 10px;
+  display: flex;
+  gap: 10px;
+}
+```
```
Comment (kaycebasques) [inline src/site/content/en/blog/flexbox-gap/index.md:32]: This code block seems out-of-place. I think we want it before the `Browser compatibility` section?

image

File: src/site/content/en/blog/flexbox-gap/index.md [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+
+{% Aside 'key-term' %}
+`gap` is the spacing *between children*. You may have heard of this type of spacing being called "gutters" or "alleys". It's space only where the children box edges touch.
+{% endAside %}
+
+`gap` is [flow relative](https://www.w3.org/TR/css-logical-1/#intro), meaning it changes dynamically based on the direction of content flow. For example, `gap` will automatically adjust for the different`writing-mode` or `direction` values that you set for your international users. This significantly eases the burden of spacing challenges for the component and CSS author. **Less code scaling further.**
+
+<figure class="w-figure">
+  <video controls autoplay loop muted class="w-screenshot">
+    <!-- <source src="https://storage.googleapis.com/web-dev-assets/portals_vp9.webm" type="video/webm; codecs=vp8"> -->
```
Comment (kaycebasques) [inline src/site/content/en/blog/flexbox-gap/index.md:44]: Do you need me to upload this? Link me to the webm file on Drive if so, plz

File: src/site/content/en/blog/flexbox-gap/index.md [hunk 6]
```diff
@@ -0,0 +1,222 @@
...
+.layout > * + * {
+  margin-block-end: 10px; 
+  margin-inline-end: 5%; 
+}
+```
+
+The above are not a full replacement for `gap` though, and often need `@media` query adjustments to account for wrapping scenarios or writing modes or direction. Adding 1 or 2 media queries doesn't seem so bad, but they can add up and lead to complicated layout logic.
+
+What the above author really intended was to have none of the child items touch. 
+s
```
Comment (kaycebasques) [inline src/site/content/en/blog/flexbox-gap/index.md:160]:
```suggestion

```

File: src/site/content/en/blog/flexbox-gap/index.md [hunk 7]
```diff
@@ -0,0 +1,222 @@
...
+
+### Previous Attempts
+
+Example #1 - pseudo-class selectors
+```css
+.layout > :not(:first-child) {
+  margin-block-end: 10px;
+  margin-inline-end: 5%;
+}
+```
```
Comment (kaycebasques) [inline src/site/content/en/blog/flexbox-gap/index.md:147]: How about styling this with the `{% Compare 'worse' %}` component? You can give it a custom label if you want to use a gentler word. https://web.dev/handbook/web-dev-components/#compare-with-custom-labels

File: src/site/content/en/blog/flexbox-gap/index.md [hunk 8]
```diff
@@ -0,0 +1,222 @@
...
+
+Example #2 - lobotomized owl selector
+```css
+.layout > * + * {
+  margin-block-end: 10px; 
+  margin-inline-end: 5%; 
+}
+```
```
Comment (kaycebasques) [inline src/site/content/en/blog/flexbox-gap/index.md:155]: Same comment as example 1

File: src/site/content/en/blog/flexbox-gap/index.md [hunk 9]
```diff
@@ -0,0 +1,222 @@
...
+DevTools supports both `grid-gap` and `gap`, this is because `gap` is essentially an alias to the previous syntaxes. 
+
+
+## New layout potential
+
+With `flex gap`, we unlock more than convenience, we unlock powerful, perfectly spaced, intrinsic layouts. In the video and following code sample below, `grid` cannot achieve the layout that `flex` can. `grid` must have equal rows and columns, even if they're intrinsically assigned. 
+
+<figure class="w-figure">
+  <video controls autoplay loop muted class="w-screenshot">
+    <!-- <source src="https://storage.googleapis.com/web-dev-assets/portals_vp9.webm" type="video/webm; codecs=vp8"> -->
```
Comment (kaycebasques) [inline src/site/content/en/blog/flexbox-gap/index.md:194]: Link me to the Drive file if you need me to upload

File: src/site/content/en/blog/flexbox-gap/index.md [hunk 10]
```diff
@@ -0,0 +1,222 @@
+---
+title: Chromium Lands Flexbox `gap`
```
Comment (kaycebasques) [inline src/site/content/en/blog/flexbox-gap/index.md:2]:
```suggestion
title: Chromium lands Flexbox `gap`
```
We use sentence case for all titles

File: src/site/content/en/blog/flexbox-gap/index.md [hunk 11]
```diff
@@ -0,0 +1,222 @@
...
+subhead: Chromium's CSS Flexbox and Multi-Column layout engines now support gap. `grid-gap` is old news. 
+authors:
+  - adamargyle
+  # - dgrogan
+description: Chromium's CSS Flexbox and Multi-Column layout engines now support gap. `grid-gap` is old news. 
+date: 2020-04-30
+updated: 2020-04-30
```
Comment (kaycebasques) [inline src/site/content/en/blog/flexbox-gap/index.md:9]:
```suggestion
```
I think we can omit this until we actually update the guide

File: src/site/content/en/blog/flexbox-gap/index.md (hunk 5 above)
Comment (argyleink) [reply src/site/content/en/blog/flexbox-gap/index.md:44]: i can upload, i need to convert

Comment (argyleink) [reply src/site/content/en/blog/flexbox-gap/index.md:43]: unless you wanna download the mp4, convert, and upload for me? i'm equipped for the task, but happy to get help?

Comment (kaycebasques) [reply src/site/content/en/blog/flexbox-gap/index.md:43]: I tried to halp but `ffmpeg` is taking forever on my machine

**Comment by jpmedley** [comment]
@kaycebasques and @argyleink I talked to David Grogan. This has a possibility of shipping in Chrome 84, so I've added appropriate labels. Since we don't know for sure, can you guys hold this until close to the Chrome beta data?

File: src/site/content/en/blog/flexbox-gap/index.md (hunk 3 above)
Comment (argyleink) [reply src/site/content/en/blog/flexbox-gap/index.md]: found one 👍

File: src/site/content/en/blog/flexbox-gap/index.md (hunk 4 above)
Comment (argyleink) [reply src/site/content/en/blog/flexbox-gap/index.md:25]: done 👍

File: src/site/content/en/blog/flexbox-gap/index.md (hunk 7 above)
Comment (argyleink) [reply src/site/content/en/blog/flexbox-gap/index.md:146]: done 👍

**Comment by kaycebasques** [comment]
@argyleink I think the only remaining thing is to upload webm versions of the videos. Please re-assign to me when that's done.

**Comment by argyleink** [comment]
cool, so this is ready to go when the feature ships. let me know if anything else needs nudged before launch.

File: src/site/content/en/blog/flexbox-gap/index.md [hunk 12]
```diff
@@ -136,24 +134,35 @@ Set both rows and columns **separately** at once
...
 
-The above are not a full replacement for `gap` though, and often need `@media` query adjustments to account for wrapping scenarios or writing modes or direction. Adding 1 or 2 media queries doesn't seem so bad, but they can add up and lead to complicated layout logic.
+{% CompareCaption %}
+[Source](https://alistapart.com/article/axiomatic-css-and-lobotomized-owls/)
+{% endCompareCaption %}
+
+{% endCompare %}
+</div>
+
+The above are not a full replacement for `gap` though, and often need `@media` query adjustments to account for wrapping scenarios, writing modes or direction. Adding 1 or 2 media queries doesn't seem so bad, but they can add up and lead to complicated layout logic.
```
Comment (jpmedley) [inline src/site/content/en/blog/flexbox-gap/index.md:161]:
```suggestion
The above are not a full replacement for `gap` though, and often need `@media` query adjustments to account for wrapping scenarios, writing modes or direction. Adding one or two media queries doesn't seem so bad, but they can add up and lead to complicated layout logic.
```

File: src/site/content/en/blog/flexbox-gap/index.md [hunk 13]
```diff
@@ -0,0 +1,223 @@
+---
+title: Chromium lands Flexbox `gap`
+subhead: Chromium's CSS Flexbox and Multi-Column layout engines now support `gap`. `grid-gap` is old news. 
+authors:
+  - adamargyle
+  - dgrogan
+description: Chromium's CSS Flexbox and Multi-Column layout engines now support `gap`. `grid-gap` is old news. 
+date: 2020-04-30
+tags:
+  - layout
```
Comment (kaycebasques) [inline src/site/content/en/blog/flexbox-gap/index.md:10]:
```suggestion
tags:
  - blog
  - layout
```

File: src/site/content/en/blog/flexbox-gap/index.md [hunk 14]
```diff
@@ -0,0 +1,224 @@
+---
+title: Chromium lands Flexbox `gap`
+subhead: Chromium's CSS Flexbox and Multi-Column layout engines now support `gap`. `grid-gap` is old news. 
+authors:
+  - adamargyle
+  - dgrogan
+description: Chromium's CSS Flexbox and Multi-Column layout engines now support `gap`. `grid-gap` is old news. 
+date: 2020-04-30
```
Comment (kaycebasques) [inline src/site/content/en/blog/flexbox-gap/index.md:8]:
```suggestion
date: 2020-05-06
```

**Comment by argyleink** [comment]
Cool, I'll make sure to update the date before merge! 

Before that though, I need @jpmedley to approve, his review is currently marked as "change requested". That'll unlock me being able to merge.

**Comment by argyleink** [comment]
doh! i think this should have waited until flex gap was in stable right? and the post date wasnt updated yet either. best way to move forward? make a patch PR that sets a draft status?

**Comment by jpmedley** [comment]
I got carried away and merged this. The date fix is in #2820

**Comment by jpmedley** [comment]
The article says 'planned for' not 'will ship', so we're good on that front.

**Pull request merged**